controllers/clustergroupupgrade_controller.go

```diff
@@ -1807,9 +1807,32 @@ func (r *ClusterGroupUpgradeReconciler) SetupWithManager(mgr ctrl.Manager) error
 	})
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&ranv1alpha1.ClusterGroupUpgrade{}).
-		Owns(placementRuleUnstructured).
-		Owns(placementBindingUnstructured).
-		Owns(policyUnstructured).
+		For(&ranv1alpha1.ClusterGroupUpgrade{}).WithEventFilter(predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
```

Comment:
just a nit:  I feel it's more clear if WithEventFilter is appended at the end of the watched resources (ie. Owns(policyUnstructured).WithEventFilter()) as it applies to the policy as well :)